docs/book/mlops-stacks/data-validators/custom.md

```diff
@@ -0,0 +1,53 @@
+---
+description: How to develop a custom Data Validator
+---
+
+{% hint style="warning" %}
+**Base abstraction in progress!**
+
+We are actively working on the base abstraction for the Data Validators, which 
+will be available soon. As a result, their extension is not recommended at the 
+moment. When you are selecting a data validator for your stack, you can use 
+one of [the existing flavors](./data-validators.md#data-validator-flavors).
+
+If you need to implement your own Data Validator flavor, you can still do so,
+but keep in mind that you may have to refactor it when the base abstraction
+is updated. 
+{% endhint %}
+
+ZenML comes equipped with [Data Validator implementations](./data-validators.md#data-validator-flavors)
+that integrate a variety of data logging and validation libraries, frameworks
+and platforms. However, if you need to use a different library or service as a
+backend for your ZenML Data Validator, you can extend ZenML to provide your own
+custom Data Validator implementation.
+
+## Build your own custom data validator
+
+If you want to implement your own custom Data Validator, you can follow the
+following steps:
+
+1. Create a class which inherits from [the `BaseDataValidator` class](https://apidocs.zenml.io/latest/api_docs/data_validators/#zenml.data_validators.base_data_validator.BaseDataValidator).
+2. Define the `FLAVOR` class variable.
+3. Override one or more of the `BaseDataValidator` methods, depending on the
+capabilities of the underlying library/service that you want to integrate.
+4. (Optional) You should also provide some standard steps that others can easily
+insert into their pipelines for instant access to data validation features.
+
+Once you are done with the implementation, you can register it through the CLI 
+as:
+
+```shell
+zenml data-validator flavor register <THE-SOURCE-PATH-OF-YOUR-DATA-VALIDATOR>
+```
+
+ZenML includes a range of Data Validator implementations provided by specific
+integration modules. You can use them as examples of how you can extend the [base Data Validator class](https://apidocs.zenml.io/latest/api_docs/data_validators/#zenml.data_validators.base_data_validator.BaseDataValidator)
+to implement your own custom Data Validator:
+
+
+|                                                                                                                                                                           | Flavor         | Integration  |
```

Comment:
I just noticed we have a table like this in some of the "custom.md" pages but not others, around 50/50 right now. Probably we should make that consistent. Personally, I feel like these overviews are a bit redundant since we already have an overview of flavors in the main page of each stack component now, and if you wanted to use existing integrations as inspiration you would probably want to check the source code on github instead of the API docs anyways.

Reply:
Maybe we could add a short hint at the end of each custom.md page instead like "Check out the [code of the existing X integrations](link to github src/zenml/integrations/X) for examples of how you can extend the X base class. See the [overview of X flavors](link to overview on stack component page) for a list integrations that include a X component."

Reply:
I worked something else out, but similar to what you were suggesting.